plugins/basestate/BaseState.cc

```diff
@@ -0,0 +1,243 @@
+                        "device "
+                     << m_baseLinkScopedName;
+            return;
+        }
+
+        // Insert the pointer in the singleton handler for retrieving it in the yarp driver
+        BaseStateDataSingleton::getBaseStateDataHandler()->setBaseStateData(&(m_baseStateData));
+
+        driver_properties.put("device", "gazebo_basestate");
+        driver_properties.put("robot", m_baseLinkScopedName);
+
+        if (!m_baseStateDriver.open(driver_properties))
+        {
+            yError() << "gz-sim-yarp-basestate-system Plugin failed: error in opening yarp "
+                        "driver";
+            return;
+        }
+
+        m_deviceScopedName
+            = m_baseLinkScopedName + "/" + driver_properties.find("yarpDeviceName").asString();
+
+        if (!Handler::getHandler()->setDevice(m_deviceScopedName, &m_baseStateDriver))
+        {
+            yError() << "gz-sim-yarp-basestate-system: failed setting scopedDeviceName(="
+                     << m_deviceScopedName << ")";
+            return;
+        }
+        m_deviceRegistered = true;
+        yInfo() << "Registered YARP device with instance name:" << m_deviceScopedName;
+    }
+
+    virtual void PostUpdate(const UpdateInfo& _info, const EntityComponentManager& _ecm) override
+    {
+        bool dataAvailable = true;
+
+        math::Pose3d worldBasePose;
+        math::Vector3d worldBaseLinVel;
+        math::Vector3d worldBaseAngVel;
+        math::Vector3d worldBaseLinAcc;
+        math::Vector3d worldBaseAngAcc;
+
+        // Get the pose of the origin of the link frame in the world reference frame
+        if (dataAvailable && m_baseLink.WorldPose(_ecm).has_value())
+            worldBasePose = m_baseLink.WorldPose(_ecm).value();
+        else
+            dataAvailable = false;
+
+        // Get the velocity of the origin of the link frame in the world reference frame
+        if (dataAvailable && m_baseLink.WorldLinearVelocity(_ecm).has_value())
+            worldBaseLinVel = m_baseLink.WorldLinearVelocity(_ecm).value();
+        else
+            dataAvailable = false;
+        if (dataAvailable && m_baseLink.WorldAngularVelocity(_ecm).has_value())
+            worldBaseAngVel = m_baseLink.WorldAngularVelocity(_ecm).value();
+        else
+            dataAvailable = false;
+
+        // Get the acceleration of the center of mass of the link in the world reference frame
+        if (dataAvailable && m_baseLink.WorldLinearAcceleration(_ecm).has_value())
+            worldBaseLinAcc = m_baseLink.WorldLinearAcceleration(_ecm).value();
+        else
+            dataAvailable = false;
+        if (dataAvailable && m_baseLink.WorldAngularAcceleration(_ecm).has_value())
+            worldBaseAngAcc = m_baseLink.WorldAngularAcceleration(_ecm).value();
+        else
+            dataAvailable = false;
+
+        std::lock_guard<std::mutex> lock(m_baseStateData.mutex);
```

Comment:
As for if/else, it may be more clear if we give this lockguard its own scope, otherwise people in the future may add a line at the end of this method, without realizing that they are actually doing stuff in the scope of this lock_guard.

Reply:
Sounds good, solved in 821498f80ef0a1645033aa19cdc807b19add48a8 also in other places inside the plugin.